server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisRateLimitingKeyFactory.scala

```diff
@@ -0,0 +1,30 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.rate.limiter.redis
+
+import org.apache.james.rate.limiter.api.RateLimitingKey
+
+class RedisRateLimitingKeyFactory(prefix: String) {
+  def from(value: String): RateLimitingKey = RedisKey(prefix + value)
+}
+
+case class RedisKey(value: String) extends RateLimitingKey {
+  override def asString(): String = value
+}
```

Comment:
Why do you need to implement this on the Redis implementation side?

Reply:
I want to separate the "prefixValue" and "bodyValue".
Then "bodyValue" will help us deserialize to user email.

Reply:
The key semantic imo highly depends on the use case. Not of the implementation.

Eg include sender? Recipient? Specify the type of entities we are rate limiting on?

Reply:
A good sign that this is uneeded is that passing the test suite do not require you to implement this...